cpp/build-support/run-test.sh

```diff
@@ -124,6 +119,42 @@ function run_test() {
   fi
 }
 
+function print_coredumps() {
+  # The script expects core files relative to the build directory with unique
+  # names per test executable because of the parallel running. So the corefile
+  # patterns must be set with prefix `core.{test-executable}*`:
+  #
+  # In case of macOS:
+  #   sudo sysctl -w kern.corefile=core.%N.%P or simply core
+  # On Linux:
+  #   sudo echo "core.%e.%p" > /proc/sys/kernel/core_pattern
```

Comment:
We can use `sysctl -w` on Linux too:

```shell
sudo sysctl -w kernel.core_pattern=core.%e.%p
```

(I know that Linux support is a follow-up task.)

Reply:
Actually this PR already supports linux, and uses sysctl, just the comment says differently :)

Reply:
Yeah, I'm using sysctl in the GitHub Actions configs, updating the comment.